src/target/renesas.c

```diff
@@ -78,17 +78,142 @@ typedef enum {
+			NONE, NONE, FLASH_MF3, PNR_FRT}},
+	{PNR_SERIES_RA4M2,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08002000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20020000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA4M3,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08002000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20020000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA4E1,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08002000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20020000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA4E2,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08001000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x2000a000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA4W1,
+		{0x01010008, 44, NONE, NONE, 0x40100000, 0x40102000, NONE, NONE, NONE, NONE, 0x20000000, 0x20018000, NONE, NONE,
+			NONE, NONE, FLASH_MF3, PNR_FRT}},
+	{PNR_SERIES_RA6M1,
+		{0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x40102000, NONE, NONE, 0x1ffe0000, 0x20000000, 0x20000000,
+			0x20020000, NONE, NONE, 0x200fe000, 0x20100000, FLASH_RV40, PNR_FRT}},
+	{PNR_SERIES_RA6M2,
+		{0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x40108000, NONE, NONE, 0x1ffe0000, 0x20000000, 0x20000000,
+			0x20040000, NONE, NONE, 0x200fe000, 0x20100000, FLASH_RV40, PNR_FRT}},
+	{PNR_SERIES_RA6M3,
+		{0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x40110000, NONE, NONE, 0x1ffe0000, 0x20000000, 0x20000000,
+			0x20040000, 0x20040000, 0x20080000, 0x200fe000, 0x20100000, FLASH_RV40, PNR_FRT}},
+	{PNR_SERIES_RA6M4,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08002000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20040000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA6M5,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08002000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20080000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA6E1,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08002000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20040000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA6E2,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08001000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x2000a000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+	{PNR_SERIES_RA6T1,
+		{0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x40102000, NONE, NONE, 0x1ffe0000, 0x1fff0000, NONE, NONE, NONE,
+			NONE, NONE, NONE, FLASH_RV40, PNR_FRT}},
+	{PNR_SERIES_RA6T2,
+		{0x0100a100, 512, NONE, NONE, 0x08000000, 0x08004000, 0x010080f0, 196, NONE, NONE, 0x20000000, 0x20010000, NONE,
+			NONE, 0x28000000, 0x28000400, FLASH_RV40, PNR_FL2}},
+};
+
+renesas_ra_family_details_s renesas_ra_family_lookup(pnr_series_t series)
+{
+	renesas_ra_family_details_s details = {};
+	for (uint32_t i = 0U; i < sizeof(renesas_ra_family) / 19; i++) {
```

Comment:
```suggestion
	for (size_t i = 0; i < ARRAY_LENGTH(renesas_ra_family); ++i) {
```
use `ARRAY_LENGTH()` (aka `(sizeof(arr) / sizeof(arr[0]))`) here to prevent mistakes, it's defined in `general.h`, and `size_t` for the type